File: backend/app/schemas/booster.py
"""
Schemas pour les boosters et l'ouverture de packs.
"""

from pydantic import BaseModel, Field
from typing import Literal
from app.schemas.card import CardResponse


class BoosterResponse(BaseModel):
    id: str
    name: str
    set_id: str  # premier set (compat) — cf. set_ids pour la liste complète
    set_ids: list[str] = []
    cards_count: int
    resource_id: str = "coins"
    resource_name: str = "Pièces"
    price: int
    guaranteed_rare: bool
    description: str = ""
    cover_image_url: str = ""


class PackOpenRequest(BaseModel):
    booster_id: str
    quantity: Literal[1, 5, 10] = 1


class PackOpenResponse(BaseModel):
    """Résultat de l'ouverture de packs."""
    packs: list[list[CardResponse]]
    total_cost: int
    remaining_coins: int  # compat : solde de pièces après achat (inchangé si payé en une autre ressource)
    resource_id: str = "coins"
    resource_name: str = "Pièces"
    new_balance: int = 0  # solde de la ressource utilisée pour payer, après achat

File: web/src/types/booster.ts
export interface Booster {
    id: string;
    name: string;
    set_id: string;
    set_ids: string[];
    cards_count: number;
    resource_id: string;
    resource_name: string;
    price: number;
    guaranteed_rare: boolean;
    description: string;
    cover_image_url: string;
}

export interface PackOpenRequest {
    booster_id: string;
    quantity: 1 | 5 | 10;
}

export interface PackOpenResponse {
    packs: import("./card").Card[][];
    total_cost: number;
    remaining_coins: number;
    resource_id: string;
    resource_name: string;
    new_balance: number;
}

export interface OwnedBooster {
    booster_id: string;
    booster_name: string;
    booster_cover_url: string | null;
    quantity: number;
    /** Booster acheté via une offre à bonus (rareté garantie, chances boostées). */
    bonus_id: number | null;
    bonus_label: string | null;
    force_min_rarity_id: string | null;
    rarity_weight_multiplier: number | null;
}

export interface BuyToInventoryResponse {
    booster_id: string;
    booster_name: string;
    quantity: number;
    total_cost: number;
    resource_id: string;
    resource_name: string;
    new_balance: number;
}

export interface OpenOwnedRequest {
    booster_id: string;
    quantity: number;
    bonus_id?: number | null;
}
